Server/Forms/FrmConnections.cs

```diff
@@ -0,0 +1,115 @@
+
+        private void FrmConnections_Load(object sender, EventArgs e)
+        {
+            if (_connectClient != null)
+            {
+                this.Text = WindowHelper.GetWindowTitle("Connections", _connectClient);
+                new Core.Packets.ServerPackets.GetConnections().Execute(_connectClient);
+            }
+        }
+
+        public void AddConnectionToListview(string processName, string localaddress, string localport, string remoteaddress, string remoteport, string state)
+        {
+            try
+            {
+                ListViewItem lvi = new ListViewItem(new string[]
+                {
+                   processName, localaddress, localport, remoteaddress , remoteport,  state
+                });
+
+                lstConnections.Invoke((MethodInvoker)delegate
+                {
+                    if (!Groups.ContainsKey(state))
+                    {
+                        ListViewGroup g = new ListViewGroup(state, state);
+                        lstConnections.Groups.Add(g);
+                        Groups.Add(state, g);
+                    }
+                    lvi.Group = lstConnections.Groups[state];
+                    lstConnections.Items.Add(lvi);
+                });
+            }
+            catch (InvalidOperationException)
+            {
+            }
+        }
+
+        public void ClearListviewItems()
+        {
+            try
+            {
+                lstConnections.Invoke((MethodInvoker)delegate
+                {
+                    lstConnections.Items.Clear();
+                });
+            }
+            catch (InvalidOperationException)
+            {
+            }
+        }
+        enum ConnectionStates : byte
+        {
+            Closed = 1,
+            Listening = 2,
+            SYN_Sent = 3,
+            Syn_Recieved = 4,
+            Established = 5,
+            Finish_Wait_1 = 6,
+            Finish_Wait_2 = 7,
+            Closed_Wait = 8,
+            Closing = 9,
+            Last_ACK = 10,
+            Time_Wait = 11,
+            Delete_TCB = 12
+        }
+
+        private void FrmConnections_FormClosing(object sender, FormClosingEventArgs e)
+        {
+
+            if (_connectClient.Value != null)
+                _connectClient.Value.FrmCon = null;
+
+        }
+
+        private void refreshToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            if (_connectClient != null)
+            {
+                new Core.Packets.ServerPackets.GetConnections().Execute(_connectClient);
+            }
+        }
+
+        private void closeConnectionToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            if (_connectClient != null)
+            {
+                foreach (ListViewItem lvi in lstConnections.SelectedItems)
+                {
+                    //send local and remote ports of connection
+                    new Core.Packets.ServerPackets.DoCloseConnection(int.Parse(lvi.SubItems[2].Text),
```

Comment:
Perhaps we should add some type of comment about it... Since we have a few people that are able to contribute but are not as familiar with the project it could be a problem if someone accidentally forgot to check the length of the array.